cmd/main.go

```diff
@@ -182,18 +187,40 @@ func main() {
 		},
 	}
 
-	if watchedNsCache := getDefaultWatchedNamespacesCacheOptions(); watchedNsCache != nil {
+	// Use transformers to strip data from Secrets and ConfigMaps
+	// that are not tracked by the operator to reduce memory usage.
+	if strings.ToLower(os.Getenv("MEMORY_OPTIMIZATION_ENABLED")) != "false" {
```

Comment:
Lets use `DISABLE_MEMORY_OPTIMIZATION` env var which users should explicitly set to `true` to disable optimization

Reply:
updated